test/JDBC/README.md

```diff
@@ -1,5 +1,377 @@
-## Description
-This JDBC test framework is used to test the Babelfish Postgresql extensions
+# JDBC Test Framework for Babelfish
+The JDBC test framework for Babelfish uses the JDBC Driver for SQL Server for database connectivity and allows us to do end-to-end testing (i.e. testing of the T-SQL syntax and TDS protocol implementation) of Babelfish without the need to write any application level code.
 
-## IMPORTANT INFORMATION!!!
-### If you need to modify any java code within src/main/java, please make sure that you have Maven installed on your system and that you recompile with `mvn clean package` prior to creating your commit.
\ No newline at end of file
+## Table of Contents
+- [How to run the test framework](#how-to-run-the-test-framework)
+  - [Build Requirements](#build-requirements)
+  - [Steps to run](#steps-to-run)
+- [How to run tests against a custom Babelfish endpoint](#how-to-run-tests-against-a-custom-babelfish-endpoint)
+- [How to control which tests should run](#how-to-control-which-tests-should-run)
+- [Writing tests](#writing-tests)
+  - [Plain SQL Batch](#plain-sql-batch)
+  - [Prepare Exec statements](#prepare-exec-statements)
+  - [Stored Procedures](#stored-procedures)
+  - [Transactions](#transactions)
+  - [Cursors](#cursors)
+  - [SQL Authentication](#sql-authentication)
+  - [Cross-dialect](#cross-dialect-intermixing-queries-in-t-sql-and-plpgsql-dialect)
+  - [IMPORTANT](#important)
+- [Adding tests](#adding-tests)
+- [Reading the console output and diff](#reading-the-console-output-and-diff)
+
+## How to run the test framework
+
+### Build Requirements
+
+Java, Maven
+
```

Comment:
I would rearrange this section to a bulleted list, and include version requirements:

To run the tests, you must install:

- Java, version XXXX or later
- Maven, version XXXX or later